src/test/java/integration/EnrollmentTests.java

```diff
@@ -0,0 +1,153 @@
+package integration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.example.learningsystem.LearningSystemApplication;
+import org.example.learningsystem.domain.Course;
+import org.example.learningsystem.domain.CourseSettings;
+import org.example.learningsystem.domain.Student;
+import org.example.learningsystem.dto.course.CourseRequestDto;
+import org.example.learningsystem.dto.student.StudentRequestDto;
+import org.example.learningsystem.service.course.CourseService;
+import org.example.learningsystem.service.student.StudentService;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@SpringBootTest(classes = {LearningSystemApplication.class})
+@AutoConfigureMockMvc
+@ActiveProfiles("test")
+public class EnrollmentTests {
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private CourseService courseService;
+    @Autowired
+    private StudentService studentService;
+    @Autowired
+    private ObjectMapper objectMapper;
+    private static Course course;
+    private static Student student;
+
+    @BeforeAll
+    static void setUp() {
+        course = new Course("Test course", "Course description", BigDecimal.valueOf(100));
+        course.setSettings(CourseSettings.builder().isPublic(true).build());
+
+        student = new Student("Name", "Surname", "name@gmail.com",
+                LocalDate.of(1900, 1, 1), BigDecimal.valueOf(150));
+    }
```

Comment:
it's not recommended to share some data across different tests, so it's better either use `@BeforeEach` annotation or create test data inside of testing method